### DISCUSSION TITLE
How to access HA over app when Cloudflare Access application is in place?

### DISCUSSION THREAD

**Discussion by thetoaster** [homeassistant-apps/app-cloudflared#731] [Q&A]
I have managed to setup remote control and can access from desktop browser with the intermediate cloudflare application in place. 
Trying to access via remote in the HA android app isn't working. I get redirected to cloudflare login page, can login, but later on, the connection cannot be established in the app.
do you see a way to enable this correctly? Has anyone managed this setup to use with app as well?

![grafik](https://github.com/user-attachments/assets/f1c78c5e-b807-4788-a56c-4c665eae1abd)

**Comment by brenner-tobias** [comment]
Thanks for your question. Unfortunately, there is no simple set-up to do that, since the HA Android App does not support the login token. There are several issues on github (e.g. [here](https://github.com/home-assistant/android/issues/3205)) but to my knowledge, there has been no development. 

I suggest that you have a look at the following references that show more or less the same solution. Apart from that, there is nothing that we can do from the add-on perepctive.

References:
https://community.home-assistant.io/t/home-assistant-app-through-cloudflare-tunnel/709951
https://www.alexsilcock.net/notes/protecting-home-assistant-with-cloudflare-access-and-mtls/
https://empty.coffee/home-assistant-cloudflare-zero-trust-setup/
https://community.home-assistant.io/t/home-assistant-app-with-cloudflare-application/658188/7
https://community.home-assistant.io/t/cloudflare-access-and-the-android-app/155372
